app/src/Elements/Line.js

```diff
@@ -0,0 +1,132 @@
+import * as THREE from 'three';
+import Element from "./Element";
+import Point from "./Point";
+import { createAccordion, createAccordionItem } from "../Utils/generators";
+import { processCoord, triangleArea } from "../Utils/utils";
+
+export default class Line extends Element {
+    constructor(obj) {
+        super('line');
+        if (obj.type === 'line_equation') {
+            this.coefficients = obj.coefficients.map(el => processCoord(el));
+            this.direction = obj.direction || 'forward';
+            this.p1 = this.p2 = undefined;
+        } else {
+            if (obj.type === 'line_side') {
+                this.p1 = obj.side[0];
+                this.p2 = obj.side[1];
+            } else {
+                this.p1 = Point.fromCoords(obj.coords1);
+                this.p2 = Point.fromCoords(obj.coords2);
+            }
+            this.coefficients = [this.p2.y.sub(this.p1.y), this.p1.x.sub(this.p2.x),
+                this.p2.x.mul(this.p1.y).sub(this.p1.x.mul(this.p2.y))];
+            this.direction = 'forward';
+        }
+    }
+
+    info(name, parent, id) {
+        const newId = `${id}_data`;
+
+        let equationStr = '';
+        if (!this.coefficients[0].equals(0)) {
+            equationStr += `${this.coefficients[0].toFraction()}x`;
+        }
+        if (this.coefficients[1].compare(0) < 0) {
+            equationStr += `${this.coefficients[1].toFraction()}y`;
+        } else if (this.coefficients[1].compare(0) > 0) {
+            equationStr += `+${this.coefficients[1].toFraction()}y`;
+        }
```

Comment:
Can be separated to small function maybe?

Reply:
Got better?